src/cve/utils/intel_source_score.py

```diff
@@ -0,0 +1,117 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import logging
+import os
+
+from langchain_community.adapters.openai import ChatCompletion
+from langchain_community.chat_models import ChatOpenAI
+from langchain_core.messages import HumanMessage
+from langchain_nvidia_ai_endpoints import ChatNVIDIA
+
+from cve.data_models.config import RunConfig
+from cve.data_models.cve_intel import CveIntel
+from cve.utils import data_utils
+from cve.utils.loggers_factory import LoggingFactory
+from cve.utils.prompting import additional_intel_prompting
+
+logger = LoggingFactory.get_agent_logger(f"morpheus.{__name__}")
+
+class IntelScorer:
+    def __init__(self,
+                 run_config: RunConfig,
+                 trace_id: str):
+        self._run_config = run_config
+        self._trace_id = trace_id
+
+    def calculate_intel_score(self, intel: CveIntel) -> CveIntel:
+        if self._run_config.engine.intel_score_model.service.type == "openai":
+            llm = ChatOpenAI(client=ChatCompletion(), model_name=self._run_config.engine.intel_score_model.model_name,
+                             max_tokens=self._run_config.engine.intel_score_model.max_tokens,
+                             temperature=self._run_config.engine.intel_score_model.temperature,
+                             top_p=self._run_config.engine.intel_score_model.top_p)
+        else:
+            llm = ChatNVIDIA(**self._run_config.engine.agent.model.model_dump(exclude={"service", "type"},
```

Comment:
@tmihalac This is also needs to get taken from the `intel_score_model` group in the configuration

Reply:
Right

Reply:
Done